### ISSUE TITLE
Are pretrained models available?

### ISSUE THREAD

**Issue by mattragoza** [RMeli/gnina-torch#33]
I would love to use this module within liGAN instead of depending on the full gnina platform. Have the caffe model weights been converted to pytorch somewhere publicly available?

**Comment by RMeli** [comment]
Unfortunately, I never converted the original Caffe models weights to PyTorch. The `examples` folder contains some scripts to reproduce part of the results of your and Paul's work, but are meant mostly for testing.

However, I would be very interested in doing so (or incorporate PRs that will allow to do that). It would be a great feature, and would serve as double check that everything works as expected. Is there any recommended way of doing this Caffe to PyTorch conversion?

PS: For easier integration, I could push the package to PyPI and conda-forge, if it is of interest.

**Comment by drewnutt** [comment]
I actually have these weights files for all of the built-in models easily available. 

The built-in model weights were pulled directly from the source code of GNINA and converted to a `.caffemodel`. Then https://github.com/vadimkantorov/caffemodel2pytorch was utilized to convert the `.caffemodel` files to `.pt` for PyTorch. 

I have created PR #34 to add these weights files.

**Comment by dkoes** [comment]
I'll add that there is now a `dumpmodel` binary built when you build gnina that will export the built-in caffe model weights (but not that actual caffe model file as you can get this easily from the source code).

**Comment by RMeli** [comment]
That's fantastic, thanks @drewnutt ! I'll incorporate PR #34 and start adding the functionality to use them easily (including model ensemble, which is currently not implemented).

**Comment by mattragoza** [comment]
Awesome! Thanks, @drewnutt.

**Comment by RMeli** [comment]
I opened #35 as a follow-up. I'll link relevant PRs there. Please feel free to comment with specific needs and missing features from your point of view.

**Comment by RMeli** [comment]
@mattragoza you can now easily load the pre-trained `default2017` and `default2018` as follows:
```python
from gninatorch.gnina import load_gnina_model

model = load_gnina_model(MODEL_NAME)
```
where `MODEL_NAME` corresponds to name of the `.pt` files in `gninatorch/weights` (without the `.pt` extension); this is essentially equivalent to GNINA's `--cnn` option (but without model ensembles, which will follow).

The `dense` model is not yet supported because I haven't managed to obtain the same output as GNINA.

**Comment by RMeli** [comment]
An ensemble of models is now also available:
```python
from gninatorch.gnina import load_gnina_models

ensemble = load_gnina_models([MODEL_NAME_1, MODEL_NAME_2, ...])
````
The ensemble of models returns `log_CNNscore`, `CNNaffinity` and `CNNvariance`.

Still no `dense` model for the time being (I suspect that there might be an issue with `BatchNorm` layer weights, but I'm still investigating), so it is not currently possible to reproduce GNINA's `default` model (since it contains two `dense` models). However, you `*_default2018_ensemble` work as expected.
